File: src/cpp_example/classification/resmlp24/resmlp24_async.cpp
/**
 * @file resmlp24_async.cpp
 * @brief Resmlp24Factory asynchronous inference example
 */

#include "factory/resmlp24_factory.hpp"
#include "common/runner/async_classification_runner.hpp"

int main(int argc, char* argv[]) {
    auto factory = std::make_unique<dxapp::Resmlp24Factory>();
    dxapp::AsyncClassificationRunner<dxapp::Resmlp24Factory> runner(std::move(factory));
    return runner.run(argc, argv);
}

File: src/python_example/classification/resmlp24/resmlp24_async.py
#!/usr/bin/env python3
# Copyright (C) 2018- DEEPX Ltd. All rights reserved.
"""
Resmlp24 Asynchronous Inference Example

Usage:
    python resmlp24_async.py --model model.dxnn --image input.jpg
"""

import sys
from pathlib import Path

_module_dir = Path(__file__).parent
_v3_dir = _module_dir.parent.parent
for _path in [str(_v3_dir), str(_module_dir)]:
    if _path not in sys.path:
        sys.path.insert(0, _path)

from factory import Resmlp24Factory
from common.runner import AsyncRunner, parse_common_args


def parse_args():
    return parse_common_args("Resmlp24 Sync Inference")


def main():
    args = parse_args()
    factory = Resmlp24Factory()
    runner = AsyncRunner(factory)
    runner.run(args)


if __name__ == "__main__":
    main()
